tests/integration/test_core.py

```diff
@@ -540,3 +545,111 @@ async def test_openid_plugin(
         assert (
             "administrator" in wordpress_client.list_roles()
         ), "An launchpad OpenID account should be associated with the WordPress admin user"
+
+
+async def test_prometheus_integration(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name: str,
+    unit_ip_list: list[str],
+):
+    """
+    arrange: after WordPress charm has been deployed and relations established.
+    act: prometheus charm joins relation
+    assert: prometheus metrics endpoint for prometheus is active and prometheus has active scrape
+        targets.
+    """
+    assert ops_test.model
+    prometheus: Application = await ops_test.model.deploy(
+        "prometheus-k8s", channel="stable", trust=True
+    )
+
+    await ops_test.model.relate(application_name, prometheus.name)
+    await ops_test.model.wait_for_idle(apps=[application_name, prometheus.name], status="active")
+
+    for unit_ip in unit_ip_list:
+        requests.get(
+            f"http://{unit_ip}:{APACHE_PROMETHEUS_SCRAPE_PORT}", timeout=10
+        ).raise_for_status()
```

Comment:
Would it be better to assert on the response code? I'm guessing the intent here is that the test should fail if something other than a `2XX` response is returned?

Reply:
Refactored to assert for status rather than raising. Much clearer what this code is trying to test now. Thank you for the feedback!